root/var/www/html/freepbx/rest/modules/provisioning-v2.php

```diff
@@ -114,6 +114,76 @@
     return $response->withStatus(500);
 });
 
+$app->post('/provisioning/connectivitycheck', function (Request $request, Response $response, $args) {
+
+    $body = $request->getParsedBody();
+    if (!$body['host']) {
+        return $response->withJson(array('message' => 'missing host parameter'),400);
+    } elseif (!$body['schema']) {
+        return $response->withJson(array('message' => 'missing schema parameter'),400);
+    } elseif ($body['schema'] !== 'http' && $body['schema'] !== 'https') {
+         return $response->withJson(array('message' => 'Invalid schema provided'),400);
+    }
+
+    $ret = array(
+        'is_reachable' => FALSE,
+        'valid_certificate' => FALSE
+    );
+
+    $ip_regexp = '/^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/';
+    $fqdn_regexp = '/(?=^.{4,253}$)(^((?!-)[a-zA-Z0-9-]{1,63}(?<!-)\.)+[a-zA-Z]{2,63}$)/';
+    if (preg_match($ip_regexp,$body['host']) === 1 && $body['host'] !== '127.0.0.1') {
+        // provided host is a valid IP address
+        $ret['host_type'] = 'IP';
+    } elseif (preg_match($fqdn_regexp,$body['host']) === 1 && $body['host'] !== 'localhost.localdomain') {
+        // provided host is a valid FQDN
+        $ret['host_type'] = 'FQDN';
+    } else {
+        return $response->withJson(array('message' => 'provided host isn\'t a valid IP address or FQDN'),400);
+    }
+
+    $headers = $request->getHeaders();
+    $curl_header = array(
+        "User: ".$headers['HTTP_USER'][0],
+        "SecretKey: ".$headers['HTTP_SECRETKEY'][0],
+        "Content-Type: application/json;charset=utf-8",
+        "Accept: application/json;charset=utf-8",
+    );
+
+    // check provided host is reachable and is this machine
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, 'https://'.$body['host'].'/provisioning/check/ping');
```

Comment:
We have to test the required URI scheme

```suggestion
    curl_setopt($ch, CURLOPT_URL, "{$body['scheme']}://{$body['host']}/provisioning/check/ping");
```